apps/internal/base/base.go

```diff
@@ -207,18 +224,33 @@ func (b Client) AuthCodeURL(ctx context.Context, clientID, redirectURI string, s
 func (b Client) AcquireTokenSilent(ctx context.Context, silent AcquireTokenSilentParameters) (AuthResult, error) {
 	authParams := b.AuthParams // This is a copy, as we dont' have a pointer receiver and authParams is not a pointer.
 	authParams.Scopes = silent.Scopes
-	authParams.AuthorizationType = authority.ATRefreshToken
 	authParams.HomeaccountID = silent.Account.HomeAccountID
-
-	if s, ok := b.manager.(cache.Serializer); ok {
-		suggestedCacheKey := authParams.CacheKey(silent.IsAppCache)
-		b.cacheAccessor.Replace(s, suggestedCacheKey)
-		defer b.cacheAccessor.Export(s, suggestedCacheKey)
-	}
-
-	storageTokenResponse, err := b.manager.Read(ctx, authParams, silent.Account)
-	if err != nil {
-		return AuthResult{}, err
+	authParams.AuthorizationType = silent.AuthorizationType
+	authParams.UserAssertion = silent.UserAssertion
+
+	var storageTokenResponse storage.TokenResponse
+	var err error
+	if authParams.AuthorizationType == authority.ATOnBehalfOf {
+		if s, ok := b.pmanager.(cache.Serializer); ok {
```

Comment:
It looks like this block of code is duplicated in the `else` clause below.  Can it be moved out of the `if` so the duplication is removed?

Reply:
This block actually is using the partitioned_manager and the one in the else clause uses a storage manager. So the serializers for both of them are different as the structure of data is different.

Reply:
Ah ok I missed that earlier.